In [1]:
#importing the required packages
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import pandas_profiling
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression
from sklearn.neighbors import LocalOutlierFactor
%matplotlib inline

C:\ProgramData\Anaconda3\lib\site-packages\statsmodels\tools\_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [2]:
#Loading the required datasets
Medicare_Inpatient = pd.read_csv('Medicare_Charge_Inpatient_DRG100_DRG_Summary_by_DRG_FY2011.csv')
Medicare_Outpatient = pd.read_csv('Medicare_Charge_Outpatient_APC30_Summary_by_APC_CY2011.csv')
Medicare_Provider_Inpatient = pd.read_csv('Medicare_Provider_Charge_Inpatient_DRG100_FY2011.csv')
Medicare_Provider_Outpatient = pd.read_csv('Medicare_Provider_Charge_Outpatient_APC30_CY2011_v2.csv')

# Exploratory data anaysis

In [3]:
Medicare_Inpatient.head()

DRG Definition  Total Discharges  \
0           039 - EXTRACRANIAL PROCEDURES W/O CC/MCC             37861   
1  057 - DEGENERATIVE NERVOUS SYSTEM DISORDERS W/...             37927   
2  064 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...             68163   
3  065 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...            110983   
4  066 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...             62346   

   Average Covered Charges  Average Total Payments  Average Medicare Payments  
0              29228.55368             6902.381289                5514.435910  
1              23924.98059             6809.799325                5791.920242  
2              49186.95644            13614.616130               12415.825360  
3              30371.27399             8037.269492                6858.689088  
4              23300.06199             5762.881821                4508.202659

In [4]:
Medicare_Inpatient.describe().T.round(2)

count      mean       std       min       25%  \
Total Discharges           100.0  75531.11  66114.00  25082.00  32504.00   
Average Covered Charges    100.0  39166.13  28305.92  14999.96  21090.62   
Average Total Payments     100.0  10609.27   7927.61   3964.39   5721.03   
Average Medicare Payments  100.0   9316.33   7553.52   2852.37   4598.39   

                                50%       75%        max  
Total Discharges           49012.50  98344.75  428495.00  
Average Covered Charges    29085.65  48226.00  158866.84  
Average Total Payments      7843.15  12605.96   44763.51  
Average Medicare Payments   6559.59  11577.94   42281.29

In [5]:
Medicare_Inpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 5 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   DRG Definition             100 non-null    object 
 1   Total Discharges           100 non-null    int64  
 2   Average Covered Charges    100 non-null    float64
 3   Average Total Payments     100 non-null    float64
 4   Average Medicare Payments  100 non-null    float64
dtypes: float64(3), int64(1), object(1)
memory usage: 4.0+ KB


In [6]:
Medicare_Inpatient.isna().sum()

DRG Definition               0
Total Discharges             0
Average Covered Charges      0
Average Total Payments       0
Average Medicare Payments    0
dtype: int64

In [7]:
Medicare_Outpatient.head()

APC  Outpatient Services  \
0    0012 - Level I Debridement & Destruction               275987   
1   0013 - Level II Debridement & Destruction               434706   
2  0015 - Level III Debridement & Destruction               509576   
3             0019 - Level I Excision/ Biopsy                27860   
4            0020 - Level II Excision/ Biopsy                40642   

   Average Estimated Submitted Charges  Average Total Payments  
0                            74.202956               16.147079  
1                           196.170070               56.133853  
2                           326.503444               97.156250  
3                          2013.181972              330.908369  
4                          3161.457873              567.156154

In [8]:
Medicare_Outpatient.describe().T.round(2)

count       mean         std      min  \
Outpatient Services                   30.0  852369.00  1650457.78  6748.00   
Average Estimated Submitted Charges   30.0    1433.84     1617.85    74.20   
Average Total Payments                30.0     295.26      323.55    16.15   

                                          25%        50%        75%  \
Outpatient Services                  51639.25  308252.50  660291.25   
Average Estimated Submitted Charges    280.44     741.23    2212.66   
Average Total Payments                  81.43     164.70     385.37   

                                            max  
Outpatient Services                  8128902.00  
Average Estimated Submitted Charges     6840.64  
Average Total Payments                  1471.80

In [9]:
Medicare_Outpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 4 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   APC                                  30 non-null     object 
 1   Outpatient Services                  30 non-null     int64  
 2   Average Estimated Submitted Charges  30 non-null     float64
 3   Average Total Payments               30 non-null     float64
dtypes: float64(2), int64(1), object(1)
memory usage: 1.1+ KB


In [10]:
Medicare_Outpatient.isna().sum()

APC                                    0
Outpatient Services                    0
Average Estimated Submitted Charges    0
Average Total Payments                 0
dtype: int64

In [11]:
Medicare_Provider_Inpatient.head()

DRG Definition  Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10016   

                      Provider Name     Provider Street Address Provider City  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE        DOTHAN   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH          BOAZ   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET      FLORENCE   
3                 ST VINCENT'S EAST  50 MEDICAL PARK EAST DRIVE    BIRMINGHAM   
4     SHELBY BAPTIST MEDICAL CENTER     1000 FIRST STREET NORTH     ALABASTER   

  Provider State  Provider Zip Code  \
0             AL              36301   
1             AL              35957   
2             AL              35631   
3             AL              35235   
4             AL              35007   

  Hospital Referral Region (HRR) Description  Total Discharges  \
0                                AL - Dothan                91   
1                            AL - Birmingham                14   
2                            AL - Birmingham                24   
3                            AL - Birmingham                25   
4                            AL - Birmingham                18   

   Average Covered Charges  Average Total Payments  Average Medicare Payments  
0              32963.07692             5777.241758                4763.736264  
1              15131.85714             5787.571429                4976.714286  
2              37560.37500             5434.958333                4453.791667  
3              13998.28000             5417.560000                4129.160000  
4              31633.27778             5658.333333                4851.444444

In [12]:
Medicare_Provider_Inpatient.describe().T.round(2)

count       mean        std       min  \
Provider Id                163065.0  255569.87  151563.67  10001.00   
Provider Zip Code          163065.0   47938.12   27854.32   1040.00   
Total Discharges           163065.0      42.78      51.10     11.00   
Average Covered Charges    163065.0   36133.96   35065.37   2459.40   
Average Total Payments     163065.0    9707.48    7664.64   2673.00   
Average Medicare Payments  163065.0    8494.50    7309.47   1148.91   

                                 25%        50%        75%        max  
Provider Id                110092.00  250007.00  380075.00  670077.00  
Provider Zip Code           27261.00   44309.00   72901.00   99835.00  
Total Discharges               17.00      27.00      49.00    3383.00  
Average Covered Charges     15947.17   25245.82   43232.59  929118.91  
Average Total Payments       5234.50    7214.10   11286.40  156158.18  
Average Medicare Payments    4192.35    6158.47   10056.88  154620.82

In [13]:
Medicare_Provider_Inpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 163065 entries, 0 to 163064
Data columns (total 12 columns):
 #   Column                                      Non-Null Count   Dtype  
---  ------                                      --------------   -----  
 0   DRG Definition                              163065 non-null  object 
 1   Provider Id                                 163065 non-null  int64  
 2   Provider Name                               163065 non-null  object 
 3   Provider Street Address                     163065 non-null  object 
 4   Provider City                               163065 non-null  object 
 5   Provider State                              163065 non-null  object 
 6   Provider Zip Code                           163065 non-null  int64  
 7   Hospital Referral Region (HRR) Description  163065 non-null  object 
 8   Total Discharges                            163065 non-null  int64  
 9   Average Covered Charges                     163065 non-null  float64
 

In [14]:
Medicare_Provider_Inpatient.isna().sum()

DRG Definition                                0
Provider Id                                   0
Provider Name                                 0
Provider Street Address                       0
Provider City                                 0
Provider State                                0
Provider Zip Code                             0
Hospital Referral Region (HRR) Description    0
Total Discharges                              0
Average Covered Charges                       0
Average Total Payments                        0
Average Medicare Payments                     0
dtype: int64

In [15]:
Medicare_Provider_Outpatient.head()

APC  Provider Id  \
0  0012 - Level I Debridement & Destruction        10029   
1  0012 - Level I Debridement & Destruction        20024   
2  0012 - Level I Debridement & Destruction        30064   
3  0012 - Level I Debridement & Destruction        30088   
4  0012 - Level I Debridement & Destruction        30111   

                                       Provider Name  \
0                EAST ALABAMA MEDICAL CENTER AND SNF   
1                 CENTRAL PENINSULA GENERAL HOSPITAL   
2   UNIVERSITY OF ARIZONA MEDICAL CTR-UNIVERSIT, THE   
3                      BANNER BAYWOOD MEDICAL CENTER   
4  UNIVERSITY OF ARIZONA MEDICAL CTR- SOUTH CAM, THE   

      Provider Street Address Provider City Provider State  Provider Zip Code  \
0      2000 PEPPERELL PARKWAY       OPELIKA             AL              36801   
1          250 HOSPITAL PLACE      SOLDOTNA             AK              99669   
2  1501 NORTH CAMPBELL AVENUE        TUCSON             AZ              85724   
3    6644 EAST BAYWOOD AVENUE          MESA             AZ              85206   
4           2800 EAST AJO WAY        TUCSON             AZ              85713   

  Hospital Referral Region (HRR) Description  Outpatient Services  \
0                            AL - Birmingham                   23   
1                             AK - Anchorage                  994   
2                                AZ - Tucson                 1765   
3                                  AZ - Mesa                   20   
4                                AZ - Tucson                   22   

   Average  Estimated Submitted Charges  Average Total Payments  
0                             78.086957               21.910435  
1                            149.589749               36.623853  
2                             50.135411               14.541841  
3                            112.400000               23.736000  
4                            152.045455               16.569091

In [16]:
Medicare_Provider_Outpatient.describe().T.round(2)

count       mean        std       min  \
Provider Id                           43372.0  265410.92  152729.93  10001.00   
Provider Zip Code                     43372.0   49041.02   27497.90   1040.00   
Outpatient Services                   43372.0     587.98    3280.40     11.00   
Average  Estimated Submitted Charges  43372.0    1473.99    1837.53     12.64   
Average Total Payments                43372.0     265.43     261.26      7.30   

                                            25%       50%        75%  \
Provider Id                           140043.00  260091.0  390101.00   
Provider Zip Code                      28645.00   46342.0   73101.00   
Outpatient Services                       42.00     128.0     347.00   
Average  Estimated Submitted Charges     284.61     710.0    2102.12   
Average Total Payments                    74.63     139.3     394.21   

                                            max  
Provider Id                           670061.00  
Provider Zip Code                      99801.00  
Outpatient Services                   219042.00  
Average  Estimated Submitted Charges   32105.64  
Average Total Payments                  2062.47

In [17]:
Medicare_Provider_Outpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43372 entries, 0 to 43371
Data columns (total 11 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   APC                                         43372 non-null  object 
 1   Provider Id                                 43372 non-null  int64  
 2   Provider Name                               43372 non-null  object 
 3   Provider Street Address                     43372 non-null  object 
 4   Provider City                               43372 non-null  object 
 5   Provider State                              43372 non-null  object 
 6   Provider Zip Code                           43372 non-null  int64  
 7   Hospital Referral Region (HRR) Description  43372 non-null  object 
 8   Outpatient Services                         43372 non-null  int64  
 9   Average  Estimated Submitted Charges        43372 non-null  float64
 10  Average To

In [18]:
Medicare_Provider_Outpatient.isna().sum()

APC                                           0
Provider Id                                   0
Provider Name                                 0
Provider Street Address                       0
Provider City                                 0
Provider State                                0
Provider Zip Code                             0
Hospital Referral Region (HRR) Description    0
Outpatient Services                           0
Average  Estimated Submitted Charges          0
Average Total Payments                        0
dtype: int64

# Part 1

In [19]:
#1A)Highest cost variation for in patient
Medicare_VAR = Medicare_Provider_Inpatient[['DRG Definition','Average Total Payments']].groupby(['DRG Definition']).agg('std').add_suffix('SD')
Medicare_VAR1 = Medicare_Provider_Inpatient[['DRG Definition','Average Total Payments']].groupby(['DRG Definition']).agg('mean').add_suffix('Mean')
Medicare_VAR['Mean_Average Total Payments'] = Medicare_VAR1['Average Total PaymentsMean']

In [20]:
#Variance is calculated by dividing standard deviation with mean
Medicare_VAR['Variation'] = Medicare_VAR.apply(lambda x: x['Average Total PaymentsSD']/x['Mean_Average Total Payments'],axis =1)

In [21]:
pd.DataFrame(Medicare_VAR['Variation'].nlargest(11)).reset_index()

DRG Definition  Variation
0                203 - BRONCHITIS & ASTHMA W/O CC/MCC   0.367247
1                                     885 - PSYCHOSES   0.350176
2   698 - OTHER KIDNEY & URINARY TRACT DIAGNOSES W...   0.317143
3   286 - CIRCULATORY DISORDERS EXCEPT AMI, W CARD...   0.312705
4         189 - PULMONARY EDEMA & RESPIRATORY FAILURE   0.312178
5   057 - DEGENERATIVE NERVOUS SYSTEM DISORDERS W/...   0.303293
6   897 - ALCOHOL/DRUG ABUSE OR DEPENDENCE W/O REH...   0.295452
7       315 - OTHER CIRCULATORY SYSTEM DIAGNOSES W CC   0.293796
8   699 - OTHER KIDNEY & URINARY TRACT DIAGNOSES W CC   0.293253
9   207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATO...   0.291396
10  064 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...   0.290106

In [22]:
#1A)Highest cost variation for out patient
Medicare_VAR = Medicare_Provider_Outpatient[['APC','Average Total Payments']].groupby(['APC']).agg('std').add_suffix('SD')
Medicare_VAR1 = Medicare_Provider_Outpatient[['APC','Average Total Payments']].groupby(['APC']).agg('mean').add_suffix('Mean')
Medicare_VAR['Mean_Average Total Payments'] = Medicare_VAR1['Average Total PaymentsMean']

In [23]:
#Variance is calculated by dividing standard deviation with mean
Medicare_VAR['Variation'] = Medicare_VAR.apply(lambda x: x['Average Total PaymentsSD']/x['Mean_Average Total Payments'],axis =1)

In [24]:
pd.DataFrame(Medicare_VAR['Variation'].nlargest(11)).reset_index()

APC  Variation
0            0012 - Level I Debridement & Destruction   0.255349
1           0013 - Level II Debridement & Destruction   0.149204
2                    0203 - Level IV Nerve Injections   0.132643
3   0336 - Magnetic Resonance Imaging and Magnetic...   0.126842
4                    0206 - Level II Nerve Injections   0.118306
5          0015 - Level III Debridement & Destruction   0.109679
6                    0020 - Level II Excision/ Biopsy   0.107186
7                     0019 - Level I Excision/ Biopsy   0.106751
8              0698 - Level II Eye Tests & Treatments   0.104985
9                     0204 - Level I Nerve Injections   0.104474
10              0608 - Level 5 Hospital Clinic Visits   0.102721

In [25]:
#1B)Highest-Cost Claims by Provider for in patient
q1b = Medicare_Provider_Inpatient.groupby(by = ['Provider Name'])[['Average Medicare Payments']].sum().add_prefix("Sum_of_").round(1)
q1b.sort_values(by = ['Sum_of_Average Medicare Payments'],ascending = False).head().reset_index()

Provider Name  Sum_of_Average Medicare Payments
0   GOOD SAMARITAN HOSPITAL                         6191391.5
1  ST JOSEPH MEDICAL CENTER                         3677011.0
2      MERCY MEDICAL CENTER                         3056968.1
3        ST JOSEPH HOSPITAL                         2916051.3
4            MERCY HOSPITAL                         2739600.7

In [26]:
#1B)Highest-Cost Claims by Provider for out patient
q1b2 = Medicare_Provider_Outpatient.groupby(by = ['Provider Name'])[['Average Total Payments']].sum().add_prefix("Sum_of_").round(1)
q1b2.sort_values(by = ['Sum_of_Average Total Payments'],ascending = False).head().reset_index()

Provider Name  Sum_of_Average Total Payments
0   GOOD SAMARITAN HOSPITAL                        29877.3
1        ST JOSEPH HOSPITAL                        29530.8
2  ST JOSEPH MEDICAL CENTER                        25018.6
3            MERCY HOSPITAL                        23050.0
4      MERCY MEDICAL CENTER                        20412.3

In [27]:
#1C)Highest-Cost Claims by Region for in patient
q1c = Medicare_Provider_Inpatient.groupby(by = ['Provider State'])[['Average Medicare Payments']].sum().add_prefix("Sum_of_").round(1)
q1c.sort_values(by = ['Sum_of_Average Medicare Payments'],ascending = False).head().reset_index()

Provider State  Sum_of_Average Medicare Payments
0             CA                       150162655.6
1             NY                        97477157.5
2             TX                        94561240.4
3             FL                        85530771.8
4             IL                        66319234.8

In [28]:
#1C)Highest-Cost Claims by Region for out patient
q1c2 = Medicare_Provider_Outpatient.groupby(by = ['Provider State'])[['Average Total Payments']].sum().add_prefix("Sum_of_").round(1)
q1c2.sort_values(by = ['Sum_of_Average Total Payments'],ascending = False).head().reset_index()

Provider State  Sum_of_Average Total Payments
0             CA                       888252.3
1             TX                       820116.1
2             FL                       603540.4
3             NY                       582332.0
4             PA                       571519.1

In [29]:
#Part1D) Highest Number of Procedures and Largest Differences between Claims and Reimbursements for in patient
Medicare_Provider_Inpatient['Diff b/w Claims and Reimb'] = Medicare_Provider_Inpatient['Average Covered Charges'] - Medicare_Provider_Inpatient['Average Total Payments']
q1d = Medicare_Provider_Inpatient.groupby(by = ['DRG Definition', 'Provider Name'])[['Diff b/w Claims and Reimb']].mean().add_prefix("Mean_of_").round(1)
q1d.sort_values(by = ['Mean_of_Diff b/w Claims and Reimb'],ascending = False).head().reset_index()

DRG Definition  \
0  207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATO...   
1  207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATO...   
2  207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATO...   
3   870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS   
4   870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS   

                         Provider Name  Mean_of_Diff b/w Claims and Reimb  
0                  WASHINGTON HOSPITAL                           798910.2  
1                    STANFORD HOSPITAL                           772960.7  
2  MEADOWLANDS HOSPITAL MEDICAL CENTER                           547388.4  
3        CROZER CHESTER MEDICAL CENTER                           546649.7  
4              NORTHBAY MEDICAL CENTER                           544112.5

In [30]:
#Part1D) Highest Number of Procedures and Largest Differences between Claims and Reimbursements for out patient
Medicare_Provider_Outpatient['Diff b/w Claims and Reimb'] = Medicare_Provider_Outpatient['Average  Estimated Submitted Charges'] - Medicare_Provider_Outpatient['Average Total Payments']
q1d2 = Medicare_Provider_Outpatient.groupby(by = ['APC', 'Provider Name'])[['Diff b/w Claims and Reimb']].mean().add_prefix("Mean_of_").round(1)
q1d2.sort_values(by = ['Mean_of_Diff b/w Claims and Reimb'],ascending = False).head().reset_index()

APC                      Provider Name  \
0  0074 - Level IV Endoscopy Upper Airway   FORT WALTON BEACH MEDICAL CENTER   
1  0074 - Level IV Endoscopy Upper Airway     SHASTA REGIONAL MEDICAL CENTER   
2  0074 - Level IV Endoscopy Upper Airway     MEMORIAL HOSPITAL JACKSONVILLE   
3         0019 - Level I Excision/ Biopsy  CENTINELA HOSPITAL MEDICAL CENTER   
4  0074 - Level IV Endoscopy Upper Airway          RIDEOUT MEMORIAL HOSPITAL   

   Mean_of_Diff b/w Claims and Reimb  
0                            30709.2  
1                            30246.5  
2                            26216.2  
3                            26073.8  
4                            24401.0

# Part 2 : Three providers least like others and three regions least like others

In [31]:
#Provider least like others for in patient category
q2a = Medicare_Provider_Inpatient.groupby(by = ['Provider Id'])[['Average Total Payments']].mean().reset_index()
q2a.head()

Provider Id  Average Total Payments
0        10001             8749.029587
1        10005             6812.135550
2        10006             8197.242449
3        10007             4860.834035
4        10008             5898.141656

In [32]:
model = LinearRegression()
model.fit(q2a['Provider Id'].values.reshape(-1,1),q2a['Average Total Payments'].values.reshape(-1,1))

LinearRegression()

In [33]:
q2a['Predicted'] = model.predict(q2a['Provider Id'].values.reshape(-1,1))
q2a['Residual'] = abs(q2a['Predicted'] - q2a['Average Total Payments'])

In [34]:
#Three Providers least like others for in patient category
q2a.sort_values('Residual',ascending=False).head(3).reset_index()

index  Provider Id  Average Total Payments    Predicted      Residual
0   3143       490142            50552.611110  8760.865700  41791.745410
1    540        60096            42446.711540  9544.225357  32902.486183
2   3024       450880            33700.882773  8832.384252  24868.498522

## The 3 providers that are least like others in the in patient category are : 
#### 1) UVA HEALTH SCIENCES CENTER - 490142
#### 2) NORTH KANSAS CITY HOSPITAL - 60096
#### 3) BAYLOR SURGICAL HOSPITAL AT FORT WORTH - 450880

In [35]:
#Provider least like others for out patient category
q2a1 = Medicare_Provider_Outpatient.groupby(by = ['Provider Id'])[['Average Total Payments']].mean().reset_index()
q2a1.head()

Provider Id  Average Total Payments
0        10001              285.699426
1        10005              254.564837
2        10006              221.871666
3        10007              251.371890
4        10008              207.481575

In [36]:
model = LinearRegression()
model.fit(q2a1['Provider Id'].values.reshape(-1,1),q2a1['Average Total Payments'].values.reshape(-1,1))

LinearRegression()

In [37]:
q2a1['Predicted'] = model.predict(q2a1['Provider Id'].values.reshape(-1,1))
q2a1['Residual'] = abs(q2a1['Predicted'] - q2a1['Average Total Payments'])

In [38]:
#Three Providers least like others for out patient category
q2a1.sort_values('Residual',ascending=False).head(3)

Provider Id  Average Total Payments   Predicted    Residual
432         50714              960.595632  275.415810  685.179823
3134       670061              712.396480  252.954963  459.441517
1474       240069              629.085289  268.548781  360.536508

## The 3 providers that are least like others in out patient category are : 
#### 1) SUTTER MATERNITY & SURGERY CENTER OF SANTA CRUZ - 50714
#### 2) SOUTH TEXAS SURGICAL HOSPITAL - 670061
#### 3) OWATONNA HOSPITAL - 240069

In [39]:
#Regions least like others for in patient category
q2b = Medicare_Provider_Inpatient.groupby(by = ['Provider State'])[['Average Total Payments']].mean().reset_index()
q2b['samp_col'] = q2b.index
q2b.head()

Provider State  Average Total Payments  samp_col
0             AK            14572.395796         0
1             AL             7568.236166         1
2             AR             8019.252815         2
3             AZ            10154.532272         3
4             CA            12629.672379         4

In [40]:
model = LinearRegression()
model.fit(q2b['samp_col'].values.reshape(-1,1),q2b['Average Total Payments'].values.reshape(-1,1))

LinearRegression()

In [41]:
q2b['Predicted'] = model.predict(q2b['samp_col'].values.reshape(-1,1))
q2b['Residual'] = abs(q2b['Predicted'] - q2b['Average Total Payments'])

In [42]:
#Three Regions least like others for in patient category
q2b.sort_values('Residual',ascending=False).head(3)

Provider State  Average Total Payments  samp_col     Predicted     Residual
0              AK            14572.395796         0  10186.036831  4386.358965
7              DC            12998.033466         7  10082.477645  2915.555822
11             HI            12775.743473        11  10023.300967  2752.442506

In [43]:
#Regions least like others for out patient category
q2b1 = Medicare_Provider_Inpatient.groupby(by = ['Provider State'])[['Average Total Payments']].mean().reset_index()
q2b1['samp_col'] = q2b1.index
q2b1.head()

Provider State  Average Total Payments  samp_col
0             AK            14572.395796         0
1             AL             7568.236166         1
2             AR             8019.252815         2
3             AZ            10154.532272         3
4             CA            12629.672379         4

In [44]:
model = LinearRegression()
model.fit(q2b1['samp_col'].values.reshape(-1,1),q2b1['Average Total Payments'].values.reshape(-1,1))

LinearRegression()

In [45]:
q2b1['Predicted'] = model.predict(q2b1['samp_col'].values.reshape(-1,1))
q2b1['Residual'] = abs(q2b1['Predicted'] - q2b1['Average Total Payments'])

In [46]:
#Three Regions least like others in out patient category
q2b1.sort_values('Residual',ascending=False).head(3)

Provider State  Average Total Payments  samp_col     Predicted     Residual
0              AK            14572.395796         0  10186.036831  4386.358965
7              DC            12998.033466         7  10082.477645  2915.555822
11             HI            12775.743473        11  10023.300967  2752.442506

## I was stuck at the part - 3 of this case study. I've tried it many times but was not able to arrive at a solution. So when i mailed A labs asking for technique to solve part - 3, i was told that its enought till part 2 and was told to leave the 3rd part and submit the case study. In case of minor mistakes please accept the submission. I will try to rectify them next time.